Source/Core/Core/FreeLookConfig.cpp

```diff
@@ -42,6 +43,10 @@ void Config::Refresh()
   }
 
   camera_config.control_type = ::Config::Get(::Config::FL1_CONTROL_TYPE);
+#ifdef USE_RETRO_ACHIEVEMENTS
+  if (::Config::Get(::Config::RA_HARDCORE_ENABLED))
+    ::Config::SetBaseOrCurrent(::Config::FREE_LOOK_ENABLED, false);
+#endif  // USE_RETRO_ACHIEVEMENTS
```

Comment:
I think this would be better off expressed as `enabled = ::Config::Get(::Config::FREE_LOOK_ENABLED)&& !::Config::Get(::Config::RA_HARDCORE_ENABLED);`. By setting the config option with SetBaseOrCurrent() you're disabling freelook for later sessions that may not actually be in hardcore mode, so the user would have to go and manually re-enable it.